### ISSUE TITLE
Allow rules to assert that no output is produced when using RuleTester

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#8157]
When using RuleTester, there is currently no way to assert that a particular test case produces no output. This is a common use-case for rules that need to avoid producing output under some circumstances. As a workaround, rules can assert that `output` is the same as `code`.

```js
ruleTester.run('foo', rule, {
    invalid: [
        {

            // No fix for this case
            code: "foo",

            // Asserts that the "output" is the same as the input
            output: "foo",
            errors: [x, y]
        }
    ]
});
```

This has a few disadvantages:

* It can't distinguish between (a) the case where no output is produced, and (b) the case where the rule produces output which happens to be identical to the input. This could cause a bug in a rule to go unnoticed if a rule produces identical output when it's not supposed to produce any output. (A bug like this would have the effect of significantly hurting autofix performance, because the source text keeps getting parsed until the multipass limit is reached. Additionally, it would prevent other fixes from being applied in the given range.)
* It's unintuitive -- in order to assert that *no* output is produced, I have to provide an `output` key with code.

---

To fix this issue, I think rules should be able to provide `null` as the `output` property.

```js
ruleTester.run('foo', rule, {
    invalid: [
        {

            // No fix for this case
            code: "foo",
            output: null,
            errors: [x, y]
        }
    ]
});
```

`output: null` would assert that none of the errors reported for this test case contain any fix suggestions.
